### PR TITLE
Import and export word as a csv file

### PR DESCRIPTION
**Pull request by haonh**

### CODE REVIEW

File: app/controllers/admin/words_controller.rb [hunk 1]
```diff
@@ -61,4 +66,18 @@ def init_word
   def word_params
     params.require(:word).permit :content, :category_id, answers_attributes: [:id, :content, :status]
   end
+
+  def export_csv
+    headers = %w{category_name word answer answer_status}
```
Comment (ttuan) [inline app/controllers/admin/words_controller.rb]: move to Settings

File: app/controllers/admin/imports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  private
+  def import_csv_file file
+    CSV.open(file.path, headers: true).each do |row|
+      category_name = row["category_name"]
+      word_content = row["word"]
+      answer = row["answer"]
+      answer_status = row["answer_status"]
+      category = save_data(Category, {name: category_name} , {name: category_name})
```
Comment (ttuan) [inline app/controllers/admin/imports_controller.rb]: ident

File: app/controllers/admin/imports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      word = save_data(Word, {content: word_content},{content: word_content, category_id: category.id})
```
Comment (ttuan) [inline app/controllers/admin/imports_controller.rb]: Add space

File: app/views/admin/words/index.html.erb [hunk 4]
```diff
@@ -15,6 +29,9 @@
         <% end %>
       </tbody>  
     </table>
+    <a href="<%= admin_words_path(format: "csv") %>" class="btn btn-primary">
+      <span class="glyphicon glyphicon-download-alt"></span><%= t("download") %>
```
Comment (ttuan) [inline app/views/admin/words/index.html.erb]: remove ()

File: app/controllers/admin/imports_controller.rb (hunk 2 above)
Comment (VuongLong) [inline app/controllers/admin/imports_controller.rb]: remove space

File: app/controllers/admin/imports_controller.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      word = save_data(Word, {content: word_content}, {content: word_content, category_id: category.id})
+      save_data(Answer, {content: answer, word_id: word.id},
+        {word_id: word.id, content: answer, status: answer_status})
+    end
+  end
+
+  def save_data model, check_fields, data
+    existedModel = model.find_by check_fields
+    if existedModel
+      existedModel.update_attributes data
```
Comment (ThuBM) [inline app/controllers/admin/imports_controller.rb]: check if update fails???

**Comment by ThuBM** [comment]
OK

**Pull request merged**